### PR TITLE
Persist KCM replicas during creation if hibernation is enabled

### LINKED ISSUE #4463: Failed to hibernate shoot clusters in creation phase
**How to categorize this issue?**

/area control-plane
/kind bug
/priority 3

**What happened**:
The hibernation of a shoot cluster failed because of the following reason:

```
"time=\"2021-08-03T05:58:49Z\" level=info msg=\"waiting until there are no running Pods in the shoot cluster... there is still at least one running Pod in the shoot cluster: kube-system/apiserver-proxy-kkrlb\" operation=reconcile shoot=garden-xxxxx\n"
```

Pods were still running in the cluster but:
```
 g k -n kube-system get po
NAME                          READY   STATUS    RESTARTS   AGE
apiserver-proxy-kkrlb         2/2     Running   0          116m
calico-node-bfjtf             1/1     Running   0          113m
csi-driver-node-bzkw6         3/3     Running   0          116m
kube-proxy-rvh8j              2/2     Running   0          116m
node-exporter-qtbsp           1/1     Running   0          116m
node-problem-detector-gchw6   1/1     Running   0          116m
g k -n kube-system get no
No resources found
```

--> _Problem_: KCM is already scaled to `0` and cannot clean up remaining pods.

**What you expected to happen**:
Hibernation operation to be successful.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create cluster and let it fail during creation
2. Trigger hibernation

**Anything else we need to know?**:
Shoot creation has never completed and when the shoot is hibernated during creation, KCM gets a `replicas: 0`. See [this comment](https://github.com/gardener/gardener/blob/9698a4a00c74d01c3d7ae31f2aaf80f5f4886321/pkg/operation/botanist/kubecontrollermanager.go#L101)

**Environment**:

- Gardener version: `v1.28.1`

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area control-plane
/kind bug

**What this PR does / why we need it**:
If hibernation is enabled during creation phase the already available replicas of the kube-controller-manager will be kept, instead of being reduced to 0. This is necessary to prevent an unwanted scale down of the kube-controller-manager if the cluster is initially created with hibernation disabled, but later on hibernation is enabled and for some reason (e.g an error or a restart of gardenlet) the create operation is restarted.

**Which issue(s) this PR fixes**:
Fixes #4463 

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
Keep the already available replicas of kube-controller-manager (if any) during Create operations regardless of whether hibernation is enabled or not.
```

### CODE REVIEW

File: pkg/operation/botanist/kubecontrollermanager_test.go [hunk 1]
```diff
@@ -201,12 +201,28 @@ var _ = Describe("KubeControllerManager", func() {
 
 				It("hibernation enabled", func() {
 					botanist.Shoot.HibernationEnabled = true
-
+					kubernetesClient.EXPECT().Client().Return(c)
+					c.EXPECT().Get(ctx, kutil.Key(namespace, "kube-controller-manager"), gomock.AssignableToTypeOf(&appsv1.Deployment{})).DoAndReturn(func(_ context.Context, _ types.NamespacedName, obj *appsv1.Deployment) error {
```
Comment (timuthy) [inline pkg/operation/botanist/kubecontrollermanager_test.go:206]: Can you add another test where the `kube-controller-deployment` is not found? This should be the default case if you create a new shoot with activated hibernation, right?

**Review by timuthy — COMMENTED** [review]
Looks good, thanks for the fix @plkokanov. I added one suggestion in the test section.

File: pkg/operation/botanist/kubecontrollermanager_test.go (hunk 1 above)
Comment (plkokanov) [reply pkg/operation/botanist/kubecontrollermanager_test.go:206]: Good suggestion, thanks. I'll add it in a sec.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**